docs/proposals/metadata/external-host-tags.md

```diff
@@ -0,0 +1,76 @@
+# External Host Tags collection
+
+- Authors: Massimiliano Pippi
+- Date: 2018-01-15
+- Status: review
+- [Discussion](https://github.com/DataDog/datadog-agent/pull/1053)
+
+## Overview
+
+Some checks (at this moment there are only 2 of them: [vsphere][] and [openstack][])
+expose a method with the signature `get_external_host_tags()` that's called by
+the Agent whenever it's time to build and send the metadata payload. The way the
+new Agent collects metadata has drastically changed and this feature wasn't been
+ported yet, thus such checks are currently not working.
+
+## Problem
+
+Previous versions of the Agent took advantage of the dynamic nature of Python
+and invoked the collection method [like this][agent5-collection], an alternative
+must be defined to provide the new Agent with the same feature.
+
+## Constraints
+
+Changes to the checks must be compatible with both the agents 5.x and 6.x.
+
+## Recommended Solution
+
+### Change the collection model from "Pull" to "Push"
```

Comment:
I agree 100% with the change in model, feels cleaner if you ask me.

Reply:
+1, we need this feature for some prometheus checks as well where host metadata is exposed as Prometheus metrics for the whole cluster, not just the local host. This model would be great for controlling the collection frequency at the check level.